anvil/checks.py

```diff
@@ -0,0 +1,20 @@
+"""
+checks.py
+
+Module containing checks. Checks are performed at "compile time" and ensure
+correct configuration is used before executing any actions
+
+"""
+
+from reportengine.checks import make_argcheck, CheckError
+
+@make_argcheck
+def check_trained_with_free_theory(training_context):
```

Comment:
the checks are a bit weird but basically the argcheck arguement names matter, they check the function which they decorate for matching argument and then perform checks on whatever that argument is, in this case it's the read yaml file but it could be a more complicated object.

Reply:
I think it will become clearer wiht additonal examples, I'll help with these on later PRs

Reply:
seems useful!